bzzeth/bzzeth.go

```diff
@@ -99,6 +119,215 @@ func (b *BzzEth) handleMsgFromSwarmNode(p *Peer) func(context.Context, interface
 	}
 }
 
+// handles new header hashes - strategy; only request headers that are in Kad Nearest Neighbourhood
+func (b *BzzEth) handleNewBlockHeaders(ctx context.Context, p *Peer, msg *NewBlockHeaders) {
+	p.logger.Debug("bzzeth.handleNewBlockHeaders")
+
+	// collect the addresses of blocks that are not in our localstore
+	var addresses []chunk.Address
+	for _, h := range *msg {
+		addresses = append(addresses, h.Hash.Bytes())
+		log.Trace("Received hashes ", "Header", hex.EncodeToString(h.Hash.Bytes()))
+	}
+	yes, err := b.netStore.Store.HasMulti(ctx, addresses...)
+
+	// collect the hashes of block headers we want
+	var hashes [][]byte
+	for i, y := range yes {
+		// ignore hashes already present in localstore
+		if y {
+			continue
+		}
+
+		// collect hash based on proximity
+		vhash := addresses[i]
+		if wantHeaderFunc(vhash, b.kad) {
+			hashes = append(hashes, vhash)
+		} else {
```

Comment:
This is super nit picking but call wantedHeader but if not wanted we report not in proximity. so you display more specific property than granted by `wantedHeaders` which can also be overwritten.
Generally code is simpler to interpret if it avoids these latent logical leaps.

Reply:
I am not sure i understand this comment. Are you saying not to log.Trace if header is not in proximity?